pipeline/dags/compute_hourly.py

```diff
@@ -3,24 +3,17 @@
 import airflow
 from airflow.operators import empty
 
-from dag_utils import date, marts
+from dag_utils import date, marts, notifications
 from dag_utils.dbt import (
     get_after_geocoding_tasks,
     get_before_geocoding_tasks,
     get_staging_tasks,
 )
-from dag_utils.notifications import format_failure, notify_webhook
-
-default_args = {
-    "on_failure_callback": lambda context: notify_webhook(
-        context, "mattermost", format_failure
-    )
-}
 
 with airflow.DAG(
     dag_id="compute_hourly",
     start_date=pendulum.datetime(2022, 1, 1, tz=date.TIME_ZONE),
-    default_args=default_args,
+    default_args=notifications.notify_failure_args(),
```

Comment:
une fonction et non un dict ?

Reply:
yes, il est considéré mauvaise pratique de passer un dict à un argument de fonction (car mutable) cf https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil

Donc cela permet d'avoir une "arg factory" au lieu d'un dict qui potentiellement est modifié au cours du load ou de l'exécution des DAGs. 

Par ailleurs, peut etre que plus tard on voudra composer différents types d'args par defaut, par exemple:
```
default_args=notify_failure_args() | n_retry_times(3)
```
et la forme de factory prend plus de sens et est moins dangereuse dans ce cas.

Reply:
ok

Reply:
éventuellement tu aurais pu faire  `default_args={**NOTIFY_FAILURE_ARGS}` bref